src/transaction.rs

```diff
@@ -278,4 +291,86 @@ impl Transaction {
             _ => None,
         }
     }
+
+    /// Used for accounting purposes when charging byte based fees
+    pub fn metered_bytes_size(&self) -> usize {
+        // Just use the default serialized size for now until
+        // the compressed representation for accounting purposes
+        // is defined. Witness data should still be excluded.
+        let witness_data = self
+            .witnesses()
+            .iter()
+            .map(|w| w.serialized_size())
+            .sum::<usize>();
+
+        self.serialized_size() - witness_data // Witness data size
```

Comment:
Because witness data that isn't used for signatures or contract creation bytecode _could_ be malleated by the block producer, making the user pay for in a way they didn't explicitly consent to. Paying for that witness data can be accomplished by some sort of baseline fee, as in

> Do we have some start gas cost for Create and Transact? If not we probably shold have it, in EVM it is big number of 21k gas.

Reply:
sure, producer can malleate transaction in some way but when block is sealed and send to another peer, somebody needs to pay for bytes used or it is going to be missused. Can we make it that BlockProducer pays for additional cost? And second thing can we restrict tx that has possibility to be malleable, on only owners who we trust to do that?

Reply:
Sure that is about topic malleated by third partie. What about BlockProducer paying for additional cost because, he is the one that has control over it?

Addionaly for baseline fee, there is small difference between paying for additional witness bytes and paying for start gas. Question is if there is no `gas_price` for Create do we use `bytes_price` and multiplier for fee on both Create and Script transactions for that start gas usage.

Reply:
Dump that all into issues 😂! They're all valid questions that will involve a lot of subtle discussion, but don't fundamentally change the transaction format. They're consensus rules.

Reply:
Yea but `gas_price` is removed from Create and it behaves differently from `bytes_price` in tearm of calculating start fee, shouldn't we have both of those guys inside Create and go with that, that would be a least bit of disrupting?

Reply:
I can put the fields back, which would also simplify the other issue John raised about `set_gas_price` since throwing errors there would need to be bubbled up in a number of places. 

On the other hand, if we're already taking in these breaking changes now maybe we should just rip the bandaid off and deal with it rather than splitting them across several incremental breaking releases? I'm sure we could come up with a simple solution in the tx pool to prioritize these vs scripts txs with or without a field saying "gas_price".

Reply:
Yeah, but this is not just effecting txpool, it effects how sdk is going to present this to a user, how will `bytes_price` and `gas_price` going to coralate together and be calculated and presented to user as recomendation to choose? And not just as recomendation how would user know if is it better to set high `bytes_price` and low `gas_price` or maybe they need to be similar, or maybe they depend on gas used or bytes spend. You need to mix them together but both of prices are defined by user and represent not related measurements.

There are lot of questions and details here, maybe I am missing it, but I dont see a great amount of benefit by doing it atm. If you desire to work on it, go for it, as John said this can be good research paper and i dont really have a strong aspiration to just stick with txpool, you can take it from here if you want.

Reply:
The reason I'm trying to get this figured out now is that this is blocking how we implement charging for fees. If it's too hard to figure out byte_price now, I'm worried it will be even harder in the future once everything is already built up around an alternate model.

Reply:
So after thinking a bit, I'm inclined to get the gas fields re-added to contract creation txs. Then we can use the gas to meter predicates as well (the cost would be deterministic, and can be subtracted before script execution begins). It was always odd that Bitcoin meters predicates only on bytes.

If it makes your life simpler @Voxelot as you said, then maybe we-add gas fields to contract creation? Then we can just set some simple consensus rules to start to restrict bytes price in a way that doesn't introduce any new complexity in the mempool compared to current-Ethereum.

Reply:
That makes sense to me, as not all opcodes are equal even if they're encoded to the same size.

Reply:
I'll add the fields back, although even if we use gas price for predicate fees which are determined upfront, the gas limit may still be a useless field for create transactions.